osquery/tables/system/linux/apt_sources.cpp

```diff
@@ -198,6 +210,140 @@ static void genAptSource(const std::string& source,
   }
 }
 
+Status parseDeb822Block(const std::string& input_block,
+                        std::vector<AptSource>& apt_sources) {
+  std::vector<std::string> uris;
+  std::vector<std::string> suites;
+
+  for (const auto& input_line : osquery::split(input_block, "\n")) {
+    // Skip empty lines
+    if (input_line.empty()) {
+      continue;
+    }
+
+    // Remove everything after # from the line for comments
+    auto comment_pos = input_line.find("#");
+    std::string line;
+    if (comment_pos != std::string::npos) {
+      line = input_line.substr(0, comment_pos);
+    } else {
+      line = input_line;
+    }
+
+    auto colon_pos = line.find(":");
+    if (colon_pos == std::string::npos) {
+      continue;
+    }
+
+    auto key = line.substr(0, colon_pos);
+    auto value = line.substr(colon_pos + 1, std::string::npos);
+    boost::trim(value);
+
+    if (value.empty()) {
+      continue;
+    }
+
+    if (key == "Types") {
+      auto types = osquery::split(value);
+      if (std::find(types.begin(), types.end(), "deb") == types.end()) {
+        return Status::failure("not a deb type repo");
+      }
+    }
+
+    if (key == "Enabled" && value != "on") {
+      return Status::failure("repo is not enabled");
+    }
+
+    if (key == "URIs") {
+      for (auto& uri : osquery::split(value)) {
+        if (uri.find("://") == std::string::npos) {
+          continue;
+        }
+        // Cannot have trailing slashes
+        while (uri.back() == '/') {
+          uri.pop_back();
+        }
+        uris.push_back(uri);
+      }
+    }
+
+    if (key == "Suites") {
+      for (const auto& suite : osquery::split(value)) {
+        suites.push_back(suite);
+      }
+    }
+  }
+
+  if (uris.size() == 0) {
+    return Status::failure("missing valid URIs");
+  }
+
+  if (suites.size() == 0) {
+    return Status::failure("missing Suites");
+  }
+
+  for (const auto& uri : uris) {
+    // All URLs should have protocol, checked earlier
+    auto proto_end = uri.find("://");
+    auto name = uri.substr(proto_end + 3);
+    for (const auto& suite : suites) {
+      auto suite_parts = osquery::split(suite, "/");
+      AptSource apt_source;
+
+      apt_source.base_uri = uri;
+      apt_source.name = name + ' ' + suite;
+      apt_source.cache_file.push_back(name);
+
+      if (suite_parts.size() == 1) {
+        apt_source.cache_file.push_back("dists");
+        apt_source.cache_file.push_back(suite);
+      } else {
+        apt_source.cache_file.insert(apt_source.cache_file.end(),
+                                     suite_parts.begin(),
+                                     suite_parts.end());
+      }
+
+      apt_sources.push_back(apt_source);
+    }
+  }
+
+  return Status::success();
+}
+
+void genDeb822Block(const std::string& source,
+                    const std::string& block,
+                    QueryData& results,
+                    Logger& logger) {
+  std::vector<AptSource> apt_sources;
+  parseDeb822Block(block, apt_sources);
```

Comment:
Check for the returned `Status`?

Reply:
I know if we want to return an error on invalid source, the original apt source doesn't check the return value either https://github.com/dantecatalfamo/osquery/blob/66accac3d3cea0be7f54ae69018410c94512c001/osquery/tables/system/linux/apt_sources.cpp#L209

Reply:
I would return it and log the error (when in verbose mode) so it doesn't fail silently.

Reply:
Am ok with keeping the consistency with `genAptUrl`&`aptSourceToRow`.

Reply:
I can log it, but I don't think an invalid block should return early and prevent the rest of thee table from loading